src/screens/TransactionConfirmationScreen.js

```diff
@@ -245,11 +246,12 @@ export default function TransactionConfirmationScreen() {
 
   useEffect(() => {
     const initProvider = async () => {
-      const p = isL2 ? await getProviderForNetwork(network) : web3Provider;
+      const p =
```

Comment:
oh now that we have `mainnet` set up in this provider cache, we can actually get rid of this check and just straight up use `getProviderForNetwork` here

Reply:
and can get rid of using `web3Provider` directly in this file

Reply:
Seems like you still need to do this (web3provider still here)